halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/SpringProfileFactory.java

```diff
@@ -41,6 +44,36 @@ protected void setProfile(
 
     profile.appendContents(
         yamlToString(deploymentConfiguration.getName(), profile, spectatorConfig));
+
+    if (addExtensibilityConfigs(deploymentConfiguration)) {
+      profile.appendContents(
+          yamlToString(
+              deploymentConfiguration.getName(),
+              profile,
+              getSpinnakerYaml(deploymentConfiguration)));
+    }
+  }
+
+  protected Map<String, Object> getSpinnakerYaml(DeploymentConfiguration deploymentConfiguration) {
+    Map<String, Object> spinnakerYaml = new LinkedHashMap<>();
+    Map<String, Object> extensibilityYaml = new LinkedHashMap<>();
+    Map<String, Object> extensibilityContents =
+        deploymentConfiguration.getSpinnaker().getExtensibility().toMap();
+    extensibilityContents.put(
+        "plugins-root-path", "/opt/" + this.getArtifact().toString().toLowerCase() + "/plugins");
+    extensibilityContents.put("strict-plugin-loading", false);
+    extensibilityYaml.put("extensibility", extensibilityContents);
+    spinnakerYaml.put("spinnaker", extensibilityYaml);
+    return spinnakerYaml;
+  }
+
+  protected boolean spinnakerVersionSupportsPlugins(
+      DeploymentConfiguration deploymentConfiguration) {
+    return Versions.greaterThanEqual(deploymentConfiguration.getVersion(), "1.19.4");
```

Comment:
this assumes the 1.19.4 release contains the kork fixes: https://github.com/spinnaker/kork/pull/586/files

Reply:
How will this handle cases where people are deploying things like `master-latest-unvalidated` or `release-1.19.x-latest-validated`, etc.?

Reply:
updated and added tests 👍